compiler/src/iree/compiler/Codegen/Common/BufferizeDispatchTensorLoadStore.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/Passes.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenDialect.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenOps.h"
+#include "iree/compiler/Codegen/Utils/Utils.h"
+#include "iree/compiler/Dialect/HAL/IR/HALOps.h"
+#include "iree/compiler/Dialect/TensorExt/IR/TensorExtTypes.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+
+namespace mlir::iree_compiler {
+
+#define GEN_PASS_DEF_BUFFERIZEDISPATCHTENSORLOADSTOREPASS
+#include "iree/compiler/Codegen/Common/Passes.h.inc"
+
+static Value getBufferizedSubspanOrSubview(
+    RewriterBase &rewriter, Location loc,
+    IREE::HAL::InterfaceBindingSubspanOp subspanOp, RankedTensorType tensorType,
+    SmallVector<OpFoldResult> offsets, SmallVector<OpFoldResult> sizes,
+    SmallVector<OpFoldResult> strides) {
+  auto dispatchTensorType =
+      cast<IREE::TensorExt::DispatchTensorType>(subspanOp->getResultTypes()[0]);
+  SmallVector<Value> subspanDynamicSizes = subspanOp.getDynamicDims();
+  Value subspanMemref = findOrCreateSubspanBuffer(rewriter, subspanOp);
+  SmallVector<Value> dynamicSizes;
+  std::tie(std::ignore, dynamicSizes) = decomposeMixedValues(sizes);
+  if (equalTensorShape(tensorType, dynamicSizes, dispatchTensorType,
+                       subspanDynamicSizes)) {
+    return subspanMemref;
+  }
+  MemRefType subviewMemRefType = memref::SubViewOp::inferRankReducedResultType(
+      tensorType.getShape(), cast<MemRefType>(subspanMemref.getType()), offsets,
+      sizes, strides);
+  return rewriter
+      .create<memref::SubViewOp>(loc, subviewMemRefType, subspanMemref, offsets,
+                                 sizes, strides)
+      .getResult();
+}
+
+static void
+bufferizeDispatchTensorLoad(IRRewriter &rewriter,
```

Comment:
nit: I think we prefer `RewriterBase` in functions because functions could be refactored to other places and it can be used by other rewriters like PatternRewriter, etc. In the context, people don't need to update the type to `RewriterBase`. So I'd suggest to use `RewriterBase` in the first place and it is also all you need in the function.